File: src/main/java/com/slackuserpresencemanager/WindowsStateManager.java
package com.slackuserpresencemanager;

import com.slackuserpresencemanager.slack.Presence;

/**
 * Created by Christian on 6/14/2017.
 * Project: SlackUserPresenceManager
 */
class WindowsStateManager {
    static void registerShutdownListener() {
        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
            HTTPManager.updatePresence(Presence.AWAY);
            HTTPManager.updateStatus(Main.getProperty("logged-off-message"), Main.getProperty("logged-off-emoji"));
        }));
    }
}

File: src/main/kotlin/com/slackuserpresencemanager/WindowsStateManager.kt
package com.slackuserpresencemanager

import com.slackuserpresencemanager.slack.Presence

/**
 * Created by Christian on 6/14/2017.
 * Project: SlackUserPresenceManager
 */
internal object WindowsStateManager {
    fun registerShutdownListener() {
        Runtime.getRuntime().addShutdownHook(Thread {
            HTTPManager.updatePresence(Presence.AWAY)
            HTTPManager.updateStatus(Main.getProperty("logged-off-message"), Main.getProperty("logged-off-emoji"))
        })
    }
}
